common/src/main/java/org/teiid/spring/common/SourceType.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.teiid.spring.common;
+
+/**
+ * The underlying source types for {@link ExternalSource}s
+ */
+public enum SourceType {
```

Comment:
I like this. Should the mustache files be moved here then one do not need to touch the vdb-code-generator?

Reply:
I considered that, but thought they didn't really belong in the resource files of the common module.  Since there's a base check for existence in the codegen tests, we'll be gently reminded to add them on every new source type.